flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/library/Summarization.java

```diff
@@ -226,11 +247,15 @@ public void setGroupRepresentativeId(K groupRepresentativeId) {
 		}
 
 		public VGV getVertexGroupValue() {
-			return f2;
+			return f2.isLeft() ? f2.left() : null;
 		}
 
 		public void setVertexGroupValue(VGV vertexGroupValue) {
-			f2 = vertexGroupValue;
+			if (vertexGroupValue == null) {
+				f2 = new Either.Right<>(NullValue.getInstance());
```

Comment:
An instance of `VertexGroupItem` is reused in the `VertexGroupReducer`. Here the setter is implicitly only called once in the open method. So I thought reusing the new Right<>(NullValue.getInstance()) wouldn't be a benefit.